### PR TITLE
Don't let catch/finally suppress main exception in IncrementalPublishingKafkaIndexTaskRunner

### PR DESCRIPTION
**Pull request by jon-wei**

### CODE REVIEW

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java [hunk 1]
```diff
@@ -616,12 +618,22 @@ public void onFailure(Throwable t)
...
       }
       finally {
         log.info("Persisting all pending data");
-        driver.persist(committerSupplier.get()); // persist pending data
+        try {
+          driver.persist(committerSupplier.get()); // persist pending data
+        }
+        catch (Exception e) {
+          if (caughtExceptionInner != null) {
+            caughtExceptionInner.addSuppressed(e);
```
Comment (jihoonson) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java]: `throw caughtExceptionInner`?

File: extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java [hunk 2]
```diff
@@ -706,21 +727,38 @@ public void onFailure(Throwable t)
...
+        }
+
+        toolbox.getDruidNodeAnnouncer().unannounce(discoveryDruidNode);
+        toolbox.getDataSegmentServerAnnouncer().unannounce();
       }
-      if (chatHandlerProvider.isPresent()) {
-        chatHandlerProvider.get().unregister(task.getId());
+      catch (Exception e) {
+        if (caughtExceptionOuter != null) {
+          caughtExceptionOuter.addSuppressed(e);
```
Comment (jihoonson) [inline extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java]: `throw caughtExceptionOuter`?

Comment (jon-wei) [reply extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java]: it's in the finally block, so if caughtExceptionOuter is not null it was already thrown earlier

Comment (jihoonson) [reply extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/IncrementalPublishingKafkaIndexTaskRunner.java]: Correct. Thanks.

**Pull request merged**